.github/workflows/no_conda_test.yml

```
@@ -0,0 +1,71 @@
+name: no_conda_test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  create:
+    tags:
+      - "**"
+  schedule:
+    # Run every day at 8:42am UTC.
+    - cron: "42 8 * * *"
+
+jobs:
+  test_benchopt:
+    name: Test without conda
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            version_python: 3.7
+            coverage: "true"
+          - os: ubuntu-latest
+            version_python: 3.8
+          - os: ubuntu-latest
+            version_python: 3.9
+            conda_cmd: "mamba"
+          - os: macos-latest
+            version_python: 3.8
+    env:
+      JUNIT_XML: "test-data.xml"
+      VERSION_PYTHON: ${{ matrix.version_python }}
+
+    defaults:
+      run:
+        # Need to use this shell to get cond working properly.
+        # See https://github.com/marketplace/actions/setup-miniconda#important
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: ${{ env.VERSION_PYTHON }}
+
+      - name: Install benchopt and its dependencies
+        run: python3 -m pip install -e .[test]
```

Do you need the `[test]`
```suggestion
        run: python3 -m pip install -e .
```